src/geom/face_quad4.C

```diff
@@ -243,4 +243,44 @@ Quad4::loose_bounding_box () const
 }
 
 
+
+bool
+Quad4::contains_point (const Point & p, Real tol) const
+{
+  if (Quad4::has_affine_map())
+    return Quad::contains_point(p, tol);
+
+  auto tri3_side = Elem::build(TRI3);
+  tri3_side->set_node(0) = const_cast<Node *>(&this->node_ref(0));
+  tri3_side->set_node(1) = const_cast<Node *>(&this->node_ref(1));
+  tri3_side->set_node(2) = const_cast<Node *>(&this->node_ref(3));
+
+  if (tri3_side->contains_point(p, tol))
+    return true;
+
+  tri3_side->set_node(0) = const_cast<Node *>(&this->node_ref(2));
+
+  return tri3_side->contains_point(p, tol);
+}
+
+
+bool
+Quad4::close_to_point (const Point & p, Real tol) const
+{
+  if (Quad4::has_affine_map())
+    return Quad::close_to_point(p, tol);
```

Comment:
@jwpeterson moving the discussion to here.

You said:
> I'm confused. It looks like there is no Quad::contains_point() currently? And the stuff with the Tri elements I would assume works in the affine_map case, not the other way around...

This could (and probably should) just be
```suggestion
    return this->close_to_point(p, tol);
```

Reply:
I was talking about `Quad::contains_point()`. I'm guessing it just calls `Elem::contains_point()`.